include/avif/avif.h

```diff
@@ -772,6 +795,16 @@ typedef struct avifDecoder
     // Set this via avifDecoderSetSource().
     avifDecoderSource requestedSource;
 
+    // If this is true and a progressive AVIF is decoded, avifDecoder will behave as if the AVIF
+    // is an image sequence, in that it will set imageCount to the available number of progressive
+    // frames available, and avifDecoderNextImage()/avifDecoderNthImage() will all for specific
+    // variants of a progressive image to be decoded. To distinguish between a progressive AVIF
+    // and an AVIF image sequence, inspect avifDecoder.progressiveState.
+    avifBool allowProgressive;
+
+    // --------------------------------------------------------------------------------------------
+    // Outputs
```

Comment:
I like splitting the avifDecoder struct members into three sections: Inputs, Outputs, and Internals. But the current division is not quite right.

For example, in the Output sections, `ignoreExif,` `ignoreXMP`, `imageCountLimit`, and `strictFlags` are inputs, not outputs. Perhaps they should be moved to the Inputs section?

Also, in the Internals section, `diag` is not really internal. Should we move it to the Output section?